src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -348,6 +294,106 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
             self.assertEqual(await call.initial_metadata(), ())
             self.assertEqual(await call.trailing_metadata(), ())
 
+    async def test_call_ok_awaited(self):
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                call = await continuation(client_call_details, request)
+                await call
+                return call
+
+        server_target, _ = await start_test_server()  # pylint: disable=unused-variable
+
+        async with aio.insecure_channel(server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+            response = await call
+
+            self.assertTrue(call.done())
+            self.assertFalse(call.cancelled())
+            self.assertEqual(type(response), messages_pb2.SimpleResponse)
+            self.assertEqual(await call.code(), grpc.StatusCode.OK)
+            self.assertEqual(await call.details(), '')
+            self.assertEqual(await call.initial_metadata(), ())
+            self.assertEqual(await call.trailing_metadata(), ())
+
+    async def test_call_rpcerror(self):
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                call = await continuation(client_call_details, request)
+                return call
+
+        server_target, server = await start_test_server()  # pylint: disable=unused-variable
+
+        async with aio.insecure_channel(server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+
+            await server.stop(None)
+
+            call = multicallable(messages_pb2.SimpleRequest(), timeout=0.1)
```

Comment:
0.1 second is a large amount of time, I didn't spot a sleep on both client-side and server-side. So, the RPC seems should not timeout. If this is due to https://github.com/grpc/grpc/issues/21605, we should note it down. This test case might not testing the right behavior.

Reply:
before the request is made the server is stopped, see one line above. So theoretically the timeout is reached because the server is no longer available.

Reply:
I can change the timeout to something more aggressive like `0.01`